### PR TITLE
Revert "Update kubebuilder to the latest working version"

### LINKED ISSUE #7630: `make test` setup-envtest unable to find version for linux
### Steps to reproduce

Running into this error when installing the envtest tools during `make test`. This was caught during PR checks of the unit tests

```
/bin/sh: 1: /home/runner/go/bin/setup-envtest: not found
=> Installing Kubebuilder test tools...
go: downloading sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20240522175850-2e9781e9fc60
go: downloading github.com/spf13/afero v1.6.0
go: downloading go.uber.org/zap v1.26.0
go: downloading sigs.k8s.io/yaml v1.3.0
go: downloading golang.org/x/text v0.14.0
go: downloading go.uber.org/multierr v1.10.0
=> Instructions:
=> Set environment variable KUBEBUILDER_ASSETS for tests.
=> KUBEBUILDER_ASSETS=""
unable to find a version that was supported for platform linux/amd64
KUBEBUILDER_ASSETS="" CGO_ENABLED=1 gotestsum --junitfile ./dist/unit_test/results.xml -- -v ./pkg/... -race -coverprofile ./dist/unit_test/ut_coverage.out
```

This is happening in `test.mk` where the controller-runtime package gets downloaded and used: https://github.com/radius-project/radius/blob/66aad2c07b0793a1c59088b4868a83596ca3e448/build/test.mk#L51

This seems to be isolated to linux and it's not happening for other archiectures

### Observed behavior

_No response_

### Desired behavior

We'd like the latest package to download for linux without the need for a hardcoded version. It's working locally on macos.

### Workaround

This is a blocking issue to PRs so we've hardcoded a previous version that we know works. We definitely want to remove the workaround so we don't have a hardcoded package version in our tests. There are a couple options for that.

1. There's an open issue in the controller-runtime repo to address this bug: https://github.com/kubernetes-sigs/controller-runtime/issues/2838. Once this gets fixed, we can revert the hardcoded version back to `latest` 
2. We can move this somewhere dependabot can manage it for us

### rad Version

build test in PR

### Operating system

_No response_

### Additional context

_No response_

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#12407](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/12407)

### PR DESCRIPTION
**Pull request by sk593**
Reverts radius-project/radius#7629

This was fixed as per the convo in https://github.com/kubernetes-sigs/controller-runtime/issues/2838. It looks like the binaries weren't released for the version of Kubernetes that we were using (they had up until 1.24 but we were using 1.23)

Fixes: #7630

### CODE REVIEW

File: build/test.mk [hunk 1]
```diff
@@ -49,7 +49,7 @@ test: test-get-envtools ## Runs unit tests, excluding kubernetes controller test
 .PHONY: test-get-envtools
 test-get-envtools:
 	@echo "$(ARROW) Installing Kubebuilder test tools..."
-	$(call go-install-tool,$(ENV_SETUP),sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240521074430-fbb7d370bebc)
+	$(call go-install-tool,$(ENV_SETUP),sigs.k8s.io/controller-runtime/tools/setup-envtest@latest)
```
Comment (rynowak) [inline build/test.mk:52]: Do you want to update the Kubernetes version? or do that separately?

**Review by rynowak — DISMISSED** [review]
Code change looks good!

**Comment by rynowak** [comment]
@sk593 - looks like your DCO signoff is missing, can you fix that and then I'll launch the functional tests?

File: build/test.mk (hunk 1 above)
Comment (sk593) [reply build/test.mk:52]: I think it makes sense to address them at the same time. I'll add that update to this PR

**Review by rynowak — DISMISSED** [review]
:shipit: - looks great thanks!

**Comment by rynowak** [comment]
Looks like DCO is missing.

**Comment by sk593** [comment]
> Looks like DCO is missing.

whoops, branch should be updated now

**Review by ytimocin — APPROVED** [review]

**Pull request merged**